pgmpy/tests/test_factors/test_Factor.py

```diff
@@ -259,47 +395,106 @@ def test_eq(self):
         self.assertTrue(self.phi == self.phi)
         self.assertTrue(self.phi1 == self.phi1)
 
+        self.assertTrue(self.phi5 == self.phi5)
+        self.assertFalse(self.phi5 == self.phi6)
+        self.assertTrue(self.phi6 == self.phi6)
+
     def test_eq1(self):
         phi1 = Factor(['x1', 'x2', 'x3'], [2, 4, 3], range(24))
-        phi2 = Factor(['x2', 'x1', 'x3'], [4, 2, 3], [0, 1, 2, 12, 13, 14, 3,
-                                                      4, 5, 15, 16, 17, 6, 7,
-                                                      8, 18, 19, 20, 9, 10, 11,
-                                                      21, 22, 23])
+        phi2 = Factor(['x2', 'x1', 'x3'], [4, 2, 3],
+                      [0, 1, 2, 12, 13, 14, 3,
+                      4, 5, 15, 16, 17, 6, 7,
+                      8, 18, 19, 20, 9, 10, 11,
+                      21, 22, 23])
         self.assertTrue(phi1 == phi2)
         self.assertEqual(phi2.variables, ['x2', 'x1', 'x3'])
 
-    def test_hash(self):
-        phi = Factor(['x1', 'x2'], [2, 2], [1, 2, 3, 4])
-        phi1 = Factor(['x2', 'x1'], [2, 2], [1, 3, 2, 4])
-        self.assertEqual(hash(phi), hash(phi1))
-
-        phi = Factor(['x1', 'x2', 'x3'], [2, 2, 2], range(8))
-        phi1 = Factor(['x3', 'x1', 'x2'], [2, 2, 2], [0, 2, 4, 6, 1, 3, 5, 7])
-        self.assertEqual(hash(phi), hash(phi1))
+        tup1 = ('x1', 'x2')
+        tup2 = ('x2', 'x3')
+        tup3 = ('x3', 'x4')
+        phi3 = Factor([tup1, tup2, tup3], [2, 4, 3], range(24))
+        phi4 = Factor([tup2, tup1, tup3], [4, 2, 3],
+                      [0, 1, 2, 12, 13, 14, 3,
+                      4, 5, 15, 16, 17, 6, 7,
+                      8, 18, 19, 20, 9, 10, 11,
+                      21, 22, 23])
+        self.assertTrue(phi3 == phi4)
 
-    def test_maximize1(self):
+    def test_hash(self):
+        phi1 = Factor(['x1', 'x2'], [2, 2], [1, 2, 3, 4])
+        phi2 = Factor(['x2', 'x1'], [2, 2], [1, 3, 2, 4])
+        self.assertEqual(hash(phi1), hash(phi2))
+
+        phi1 = Factor(['x1', 'x2', 'x3'], [2, 2, 2], range(8))
+        phi2 = Factor(['x3', 'x1', 'x2'], [2, 2, 2], [0, 2, 4, 6, 1, 3, 5, 7])
+        self.assertEqual(hash(phi1), hash(phi2))
+
+        class TestHash:
```

Comment:
@ankurankan But this class is only required in that function. Do I just declare it in test_Factor.py ?